src/libs/migrations/PersonalDetailsByAccountID.js

```diff
@@ -0,0 +1,213 @@
+import _ from 'underscore';
+import lodashHas from 'lodash/has';
+import Onyx from 'react-native-onyx';
+import ONYXKEYS from '../../ONYXKEYS';
+import Log from '../Log';
+
+const DEPRECATED_ONYX_KEYS = {
+    // Deprecated personal details object which was keyed by login instead of accountID.
+    PERSONAL_DETAILS: 'personalDetails',
+};
+
+/**
+ * @returns {Promise<Object>}
+ */
+function getReportActionsFromOnyx() {
+    return new Promise((resolve) => {
+        const connectionID = Onyx.connect({
+            key: ONYXKEYS.COLLECTION.REPORT_ACTIONS,
+            waitForCollectionCallback: true,
+            callback: (allReportActions) => {
+                Onyx.disconnect(connectionID);
+                return resolve(allReportActions);
+            },
+        });
+    });
+}
+
+/**
+ * We use the old personalDetails object becuase it is more efficient for this migration since it is keyed by email address.
+ * Also, if the old personalDetails object is not available, that likely means the migration has already run successfully before on this account.
+ *
+ * @returns {Promise<Object>}
+ */
+function getDeprecatedPersonalDetailsFromOnyx() {
+    return new Promise((resolve) => {
+        const connectionID = Onyx.connect({
+            key: DEPRECATED_ONYX_KEYS.PERSONAL_DETAILS,
+            callback: (allPersonalDetails) => {
+                Onyx.disconnect(connectionID);
+                return resolve(allPersonalDetails);
+            },
+        });
+    });
+}
+
+/**
+ * Migrate Onyx data for the email to accountID migration.
+ *
+ * - reportAction_
+ *     - originalMessage.oldLogin -> originalMessage.oldAccountID
+ *     - originalMessage.newLogin -> originalMessage.newAccountID
+ *     - accountEmail -> accountID
+ *     - actorEmail -> actorAccountID
+ *     - childManagerEmail -> childManagerAccountID
+ *     - whisperedTo -> whisperedToAccountIDs
+ *     - childOldestFourEmails -> childOldestFourAccountIDs
+ *     - originalMessage.participants -> originalMessage.participantAccountIDs
```

Comment:
NAB, should we rename this to participant**s**AccountIDs (backend change required) or it's too late?

Reply:
Yes, too late unfortunately. This is how the back-end is already sending, so we'd need to update the back-end and do another migration.

Reply:
IMO `participantAccountIDs` rolls off the tongue easier, also isn't it still grammatically correct? 🤔

Reply:
It just may create the confusion that we are referring to one participant with multiple account ids